src/plugins/auto_batch/src/compiled_model.cpp

```diff
@@ -154,93 +134,86 @@ std::pair<CompiledModel::WorkerInferRequest&, int> CompiledModel::GetWorkerInfer
-    if (timeout == user_config.end() || user_config.size() > 1) {
-        IE_THROW() << "The only config that can be changed on the fly for the AutoBatching the is the "
-                   << CONFIG_KEY(AUTO_BATCH_TIMEOUT);
+void CompiledModel::set_property(const ov::AnyMap& properties) {
+    auto time_out = properties.find(ov::auto_batch_timeout.name());
+    if (time_out == properties.end() || properties.size() > 1) {
+        OPENVINO_THROW("The only config that can be changed on the fly for the AutoBatching is the ",
+                       ov::auto_batch_timeout.name());
     } else {
-        m_timeout = ParseTimeoutValue(timeout->second.as<std::string>());
+        m_timeOut = time_out->second.as<std::uint32_t>();
     }
 }
 
-InferenceEngine::Parameter CompiledModel::GetConfig(const std::string& name) const {
+ov::Any CompiledModel::get_property(const std::string& name) const {
     auto it = m_config.find(name);
     if (it != m_config.end()) {
         return it->second;
     } else {
         // find config key among networks config keys
-        auto param = m_model_without_batch->GetMetric(METRIC_KEY(SUPPORTED_CONFIG_KEYS));
-        for (auto&& configKey : param.as<std::vector<std::string>>()) {
-            if (configKey == name) {
-                return m_model_without_batch->GetConfig(configKey);
+        auto modelSupportedProperties = m_compiled_model_without_batch->get_property(ov::supported_properties.name());
+        for (auto&& property : modelSupportedProperties.as<std::vector<ov::PropertyName>>()) {
+            if (property == name) {
+                return m_compiled_model_without_batch->get_property(property);
             }
         }
-        IE_THROW(NotFound) << name << " not found in the ExecutableNetwork config";
+        if (name == ov::optimal_number_of_infer_requests.name()) {
+            uint32_t num_request = 0;
+            try {
+                num_request =
+                    m_compiled_model_without_batch->get_property(ov::hint::num_requests.name()).as<std::uint32_t>();
+                if (num_request == 0)  // no limitations from user, let's deduce the full blown #requests
+                    // (multiplied by the devices capabilities to run multiple <batched> requests for further perf)
+                    num_request =
+                        m_device_info.device_batch_size *
+                        m_compiled_model_without_batch->get_property(ov::optimal_number_of_infer_requests.name())
+                            .as<uint32_t>();
+            } catch (const ov::Exception&) {
+            }
+            num_request =
+                std::max(num_request, m_device_info.device_batch_size);  // round up to the possible  user's value
+            return num_request;
+        } else if (name == ov::model_name.name()) {
+            return m_compiled_model_without_batch->get_property(name);
+            OPENVINO_SUPPRESS_DEPRECATED_START
```

Comment:
It is a little strange to put `OPENVINO_SUPPRESS_DEPRECATED_START` in this 'else if` branch.